data-products/src/main/scala/org/sunbird/analytics/job/report/CollectionSummaryJob.scala

```diff
@@ -0,0 +1,188 @@
+  }
+
+  def prepareReport(spark: SparkSession, fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame)(implicit fc: FrameworkContext, config: JobConfig): DataFrame = {
+    implicit val sparkSession: SparkSession = spark
+    implicit val sqlContext: SQLContext = spark.sqlContext
+    import spark.implicits._
+    val userCachedDF = getUserData(spark, fetchData = fetchData).select("userid", "userchannel", "userstate")
+    val processBatches: DataFrame = filterBatches(spark, fetchData, config)
+      .join(getUserEnrollment(spark, fetchData), Seq("batchid", "courseid"), "left_outer")
+      .join(userCachedDF, Seq("userid"), "left_outer")
+
+    val filteredBatches = processBatches.select(col("*"))
+      .withColumn("isCertified", when(col("certificates").isNotNull && size(col("certificates").cast("array<map<string, string>>")) > 0, "Y")
+        .when(col("issued_certificates").isNotNull && size(col("issued_certificates").cast("array<map<string, string>>")) > 0, "Y").otherwise("N"))
+    val processedBatches = computeValues(filteredBatches.drop("issued_certificates", "certificates", "userchannel"))
+    val searchFilter = config.modelParams.get.get("searchFilter").asInstanceOf[Option[Map[String, AnyRef]]];
+    val reportDF = if (searchFilter.isEmpty) {
+      val courseIds = processedBatches.select(col("courseid")).distinct().collect().map(_ (0)).toList.asInstanceOf[List[String]]
+      val courseInfo = CourseUtils.getCourseInfo(courseIds, None, config.modelParams.get.getOrElse("maxlimit", 500).asInstanceOf[Int]).toDF("framework", "identifier", "name", "channel", "batches", "organisation")
+      JobLogger.log(s"Total courseInfo records ${courseInfo.count()}", None, INFO)
+      processedBatches.join(courseInfo, processedBatches.col("courseid") === courseInfo.col("identifier"), "inner")
+        .withColumn("collectionName", col("name"))
+        .withColumn("publishedBy", concat_ws(", ", col("organisation")))
+    } else {
+      processedBatches
+    }
+    reportDF.select("publishedBy", "batchid", "courseid", "collectionName", "startdate", "enddate", "hasCertified", "userstate", "enrolledUsersCountByState", "completionUserCountByState", "certificateIssueCount")
+  }
+
+  def computeValues(transformedDF: DataFrame): DataFrame = {
+    // Compute completionCount and enrolCount for state
+    val statePartitionDF = transformedDF.groupBy("batchid", "courseid", "userstate").agg(
+      count(when(col("status") === 2, 1)).as("completionUserCountByState"),
+      count(when(col("isCertified") === "Y", 1)).as("certificateIssueCount"),
+      count(col("userid")).as("enrolledUsersCountByState")
+    )
+    statePartitionDF.join(transformedDF.drop("isCertified", "status", "userstate").dropDuplicates("courseid", "batchid"), Seq("courseid", "batchid"), "inner")
+      .withColumn("batchid", concat(lit("batch-"), col("batchid")))
+      .withColumn("hasCertified", when(col("certificateIssueCount") > 0, "Y").otherwise("N"))
```

Comment:
Shouldn't this come from content model?

Reply:
I have checked with Pradyumna, As of now in the content model, they are not storing any certificates related information.

Reply:
Marking the flag by referring course_batch table